plonky2x/core/src/frontend/eth/rlp/utils.rs

```diff
@@ -0,0 +1,379 @@
+use ethers::types::Bytes;
+use log::info;
+use num::bigint::ToBigInt;
+use num::BigInt;
+
+/// Byte array of at most 32 bytes, potentially padded. This could be a hash, value,
+/// rlp.encode(branch node), rlp.encode(leaf node), or rlp.encode(extension node) under the hood.
+pub const MAX_STRING_SIZE: usize = 32;
+pub type FixedSizeString = [u8; MAX_STRING_SIZE];
+
+/// This represents a node in a MPT, potentially padded. This could be a null, branch, leaf, or
+/// extension node. Note that a nested list (when a small node is referenced inside node), if any,
+/// isn't decoded.
+///
+/// Why we can use FixdSizeString for the type of each element in FixedSizeMPTNode:
+/// 1. For a branch node, each node is referenced by a 32-byte hash, or the node itself. We use the
+///    node if and only if the len(node) < 32.
+/// 2. For a leaf node, the encodedPath is an an array of up to 32 bytes because a path in Ethereum
+///    is exactly 64 hex character long. Furthermore, the value is guaranteed to be up to 32 bytes.
+/// 3. For an extension node, the encodedPath is the same as the leaf node. The key is keccak256(x),
+///    which is exactly 32 bytes.
+pub const MAX_NODE_SIZE: usize = 17;
+pub type FixedSizeMPTNode = [FixedSizeString; MAX_NODE_SIZE];
+
+/// This represents the length of each string in a node, potentially padded Since each
+/// FixedSizeString in FixedSizeString is padded, we need to keep track of the "true" length of each
+/// string.
+pub type FixedSizeStringLengths = [usize; MAX_NODE_SIZE];
+
+pub fn bool_to_u32(b: bool) -> u32 {
+    if b {
+        1
+    } else {
+        0
+    }
+}
```

Comment:
https://github.com/succinctlabs/optimism-bedrock-contracts/blob/main/rlp/RLPReader.sol

We should follow the conventions in this file and have RLPItem etc. for even the normal functions?